### DISCUSSION TITLE
no-typename-prefix exceptions

### DISCUSSION THREAD

**Discussion by titonobre** [graphql-hive/graphql-eslint#1119] [Q&A] [answered]
Hi.

I have the rule `no-typename-prefix` enabled and would like to keep it this way. However there are some known cases where I would like to open an exception, ideally without using `eslint-disable-next-line`.

But before, some context. I'm following some design rules from [here](https://www.apollographql.com/blog/graphql/explaining-graphql-connections/). One of them is related to the naming of the fields.

> ```gql
> type User {
>   id: ID!
>   name: String
>   friendsConnection(
>     first: Int,
>     after: String,
>     last: Int,
>     before: String
>   ): UserFriendsConnection
> }
> ```
> Note here that we name the field `${Relationship Type}Connection`. This is so that we reserve the friends field name for a future pagination specification.

Back to my scenario. I would like to implement the following types

```gql
type Categories {
  id: ID!
  title: String
  categoriesConnection: CategoriesConnection!
}

# type CategoriesConnection with a set of CategoryEdge ...

# type CategoryEdge with a reference to a Category

type Category {
  id: ID!
  title: String
  icon: String
  items: ItemsConnection!
}
```

The issue here is that `categories` is both the name of the type and the plural of `Category`.

Would it be possible to have this rule configurable with something like this?

```js
{
  ignoreSuffix: "Connection"
}
```

Thank you.

**Comment by dimaMachina** [comment]
Sorry, I’m not plan to adding this option

**Reply by titonobre** [reply]
A Pull Request would be OK for you?

**Reply by dimaMachina** [reply, marked as answer]
@titonobre I don't want to add some extra options to this rule with the explicit name `no-typename-prefix` for the following reasons:
1. there is already a possibility to ignore your specific cases with `eslint-ignore` comments
2. your design follows from an outdated article from 2017 year, maybe I would change my mind if it was Relay spec for the current year for example...
Sorry, but not.

**Reply by titonobre** [reply]
Fair enough. Thank you.
